python_modules/dagster/dagster_tests/general_tests/grpc_tests/test_metrics.py

```diff
@@ -0,0 +1,108 @@
+from dagster._serdes import deserialize_value
+from dagster._utils import (
+    file_relative_path,
+    find_free_port,
+)
+
+
+def is_grpc_method(attr_name: str, obj: Any) -> bool:
+    """Grpc methods use pascal casing. This function checks if the given attribute is a method with pascal casing."""
+    pascal_case_rgx = re.compile(r"^[A-Z][a-z]*([A-Z][a-z]*)")
+    return (
+        inspect.isfunction(getattr(obj, attr_name)) and pascal_case_rgx.match(attr_name) is not None
+    )
+
+
+def test_metrics_retrieval_annotations():
+    for attr in dir(DagsterApiServer):
+        if is_grpc_method(attr, DagsterApiServer):
+            if attr != "ExternalSensorExecution":
+                continue
+            assert attr in METRICS_RETRIEVAL_FUNCTIONS, "Missing annotation for method: " + attr
+    for attr in METRICS_RETRIEVAL_FUNCTIONS:
+        if attr != "ExternalSensorExecution":
+            continue
+        assert is_grpc_method(attr, DagsterApiServer), "Missing method for annotation: " + attr
+
+
+def _launch_sensor_execution(
+    repo_origin: ExternalRepositoryOrigin, client: DagsterGrpcClient, instance: DagsterInstance
+):
+    client.external_sensor_execution(
+        sensor_execution_args=SensorExecutionArgs(
+            repository_origin=repo_origin,
+            instance_ref=instance.get_ref(),
+            sensor_name="extremely_slow_sensor",
+            last_tick_completion_time=None,
+            last_run_key=None,
+            cursor=None,
+            timeout=2,
+            last_sensor_start_time=None,
+        ),
+    )
+
+
+def test_ping_metrics_retrieval():
+    port = find_free_port()
+    python_file = file_relative_path(__file__, "grpc_repo_sensor_eval.py")
+
+    subprocess_args = [
+        "dagster",
+        "api",
+        "grpc",
+        "--port",
+        str(port),
+        "--python-file",
+        python_file,
+    ]
+
+    process = subprocess.Popen(subprocess_args)
+
+    try:
+        wait_for_grpc_server(
+            process, DagsterGrpcClient(port=port, host="localhost"), subprocess_args
+        )
+        client = DagsterGrpcClient(port=port)
+
+        with instance_for_test() as instance:
+            repo_origin = ExternalRepositoryOrigin(
+                code_location_origin=GrpcServerCodeLocationOrigin(port=port, host="localhost"),
+                repository_name="the_repo",
+            )
+            threading.Thread(
+                target=_launch_sensor_execution, args=(repo_origin, client, instance)
+            ).start()
+            time.sleep(2)  # wait for sensor execution to begin
+            res = client.ping("blah")
+            metadata = json.loads(res["serialized_server_health_metadata"])
+            assert len(metadata["SyncExternalSensorExecution"]) == 1
+            args = deserialize_value(
+                list(metadata["SyncExternalSensorExecution"].values()).pop(), SensorExecutionArgs
+            )
+            assert args.sensor_name == "extremely_slow_sensor"
```

Comment:
this isn't so much 'metrics' as 'the entire serialized argument to the function' right? are you planning to some processing here to compute counts or other stats on this? Otherwise this could get very large very quickly (some users have 10000 sensors)

Reply:
oh i see, its what requests are in the middle of being processed at this moment, so that scopes the length somewhat

Reply:
that's right. n <= max_workers.